tslearn/svm.py

```diff
@@ -32,6 +32,29 @@ def _kernel_func_gak(sz, d, gamma):
         gamma = 1.
     return lambda x, y: cdist_gak(x.reshape((-1, sz, d)), y.reshape((-1, sz, d)), sigma=numpy.sqrt(gamma / 2.))
 
+def _sparse_kernel_func_gak(sz, d, gamma, slice_support_vectors=None):
+    if gamma == "auto":
+        gamma = 1.
+    
+    def sparse_gak(X, X_fit):
+        
+        if X_fit is X:
+            return cdist_gak(X.reshape((-1, sz, d)), None, sigma=numpy.sqrt(gamma / 2.))
+        
+        if slice_support_vectors is not None:
+            # slice out support vectors
+            sliced_X_fit = X_fit[slice_support_vectors]
+            gak_sim_dense = cdist_gak(X.reshape((-1, sz, d)), sliced_X_fit.reshape((-1, sz, d)), sigma=numpy.sqrt(gamma / 2.))
+
+            # act like nothing has happend ...
+            gak_sim = numpy.empty((len(X), len(X_fit)))
+            gak_sim[:, slice_support_vectors] = gak_sim_dense
```

Comment:
Should we document somewhere that, at predict time, only those entries corresponding to SVs will be meaningful?

Reply:
It can be mentioned but I see it more as standard behaviour for an SVM, I don't completely see why scikit-learn would not do this slicing by default for callable kernels, maybe there is a use case where this would be a problem ...

Reply:
You are right, and if I check the API for this class on sklearn, I do not see a single place where a end-user would end up playing with that matrix, so I guess keeping that warning in the code could be sufficient.